geopandas/tests/test_geodataframe.py

```diff
@@ -1165,3 +1166,13 @@ def test_geodataframe_crs():
     gdf = GeoDataFrame(columns=["geometry"])
     gdf.crs = "IGNF:ETRS89UTM28"
     assert gdf.crs.to_authority() == ("IGNF", "ETRS89UTM28")
+
+
+def test_setitem_new_index(dfs):
+    gdf = dfs[0]
+
+    gdf.loc[3, "geometry"] = None
+    assert is_geometry_type(gdf.geometry)
+    # internally two blockmanagers are concatenated here
+    gdf.loc[4] = [1, None]
+    assert is_geometry_type(gdf.geometry)
```

Comment:
Also this test seems to pass on main branch as well?

Reply:
(at least with pandas 1.4.1)

Reply:
I wrote a comment about this in the conversation view, should have done it on the review - this behaviour in a test_plotting test that was and still is failing with this PR that I'm not quite sure how to fix. But I should go through and work out exactly how this is different from the main branch and see if that gives any clues.